test/extension/numeric/convolve.cpp

```diff
@@ -110,12 +110,244 @@ struct test_image_5x5_kernel_3x3_identity
+                img_expected_row_it[x] = img_it[x - 2];
+        }
+        for (std::ptrdiff_t x = 0; x < img_view.width(); ++x)
+        {
+            auto img_expected_row_it = gil::view(img_expected_row).col_begin(x);
+            auto img_expected_col_it = gil::view(img_expected_col).col_begin(x);
+            for (std::ptrdiff_t y = 2; y < img_view.height(); ++y)
+                img_expected_col_it[y] = img_expected_row_it[y - 2];
+        }
+
+        auto const kernel = fixture::create_kernel<channel_t>({0, 0, 0, 0, 0, 0, 1, 0, 0});
+        gil::detail::convolve_1d<pixel_t>(gil::const_view(img_out), kernel, gil::view(img_out),
+            boost::gil::boundary_option::extend_zero);
+
+        BOOST_TEST(gil::equal_pixels(gil::const_view(img_out), gil::const_view(img_expected_col)));
+    }
+    static void run()
+    {
+        boost::mp11::mp_for_each<fixture::image_types>(
+            test_image_5x5_kernel_1x9_boundary_extend_zero{});
+    }
+};
+
+struct test_image_5x5_kernel_1x9_boundary_extend_constant
+{
+    template <typename Image>
+    void operator()(Image const&)
+    {
+        using image_t = Image;
+        using pixel_t = typename image_t::value_type;
+        using channel_t = typename gil::channel_type<pixel_t>::type;
+        auto img = fixture::generate_image<image_t>(5, 5, fixture::random_value<channel_t>{});
+        auto img_view = gil::view(img);
+        image_t img_out(img), img_expected_row(img);
+
+        for (std::ptrdiff_t y = 0; y < img_view.height(); ++y)
+        {
+            auto img_it = img_view.row_begin(y);
+            auto img_expected_row_it = gil::view(img_expected_row).row_begin(y);
+            for (std::ptrdiff_t x = 2; x < img_view.width(); ++x)
+                img_expected_row_it[x] = img_it[x - 2];
+            img_expected_row_it[1] = img_it[0];
+        }
+        image_t img_expected_col(img_expected_row);
+        for (std::ptrdiff_t x = 0; x < img_view.width(); ++x)
+        {
+            auto img_expected_row_it = gil::view(img_expected_row).col_begin(x);
+            auto img_expected_col_it = gil::view(img_expected_col).col_begin(x);
+            for (std::ptrdiff_t y = 2; y < img_view.height(); ++y)
+                img_expected_col_it[y] = img_expected_row_it[y - 2];
+            img_expected_col_it[1] = img_expected_row_it[0];
+        }
+
+        auto const kernel = fixture::create_kernel<channel_t>({0, 0, 0, 0, 0, 0, 1, 0, 0});
+        gil::detail::convolve_1d<pixel_t>(gil::const_view(img_out), kernel, gil::view(img_out),
+            boost::gil::boundary_option::extend_constant);
+
+        BOOST_TEST(gil::equal_pixels(gil::const_view(img_out), gil::const_view(img_expected_col)));
+    }
+    static void run()
+    {
+        boost::mp11::mp_for_each<fixture::image_types>(
+            test_image_5x5_kernel_1x9_boundary_extend_constant{});
+    }
+};
+
+struct test_image_5x5_kernel_1x9_boundary_output_zero
```

Comment:
The test uses 1x3 kernel, am I missing some context on why this is named 1x9?

Reply:
You are right, I was using 1x9 kernel in an earlier version.